core/keyserver/epochs.go

```diff
@@ -117,47 +129,87 @@ func (s *Server) GetMutations(ctx context.Context, in *pb.GetMutationsRequest) (
 		indexes = append(indexes, m.Mutation.GetIndex())
 	}
 	// Get leaf proofs.
-	// TODO: allow leaf proofs to be optional.
-	var epoch int64
-	if in.Epoch > 1 {
-		epoch = in.Epoch - 1
-	} else {
-		epoch = 1
-	}
-	proofs, err := s.inclusionProofs(ctx, in.DomainId, indexes, epoch)
+	proofs, err := s.inclusionProofs(ctx, in.DomainId, indexes, in.Epoch-1)
 	if err != nil {
 		return nil, err
 	}
 	for i, p := range proofs {
 		mutations[i].LeafProof = p
 	}
 
-	// MapRevisions start at 1. Log leave's index starts at 1.
-	// MapRevision should be at least 1 since the Signer is
-	// supposed to create at least one revision on startup.
-	respEpoch := resp.GetMapRoot().GetMapRevision()
-	// Fetch log proofs.
-	logRoot, logConsistency, logInclusion, err := s.logProofs(ctx, domain.LogID, in.GetFirstTreeSize(), respEpoch)
-	if err != nil {
-		return nil, err
-	}
-
 	nextPageToken := ""
 	if len(mutations) == int(in.PageSize) && maxSequence != highestSeq {
 		nextPageToken = fmt.Sprintf("%d", maxSequence)
 	}
-	return &pb.GetMutationsResponse{
-		Epoch:          in.Epoch,
-		Smr:            resp.GetMapRoot(),
-		LogRoot:        logRoot.GetSignedLogRoot(),
-		LogConsistency: logConsistency.GetProof().GetHashes(),
-		LogInclusion:   logInclusion.GetProof().GetHashes(),
-		Mutations:      mutations,
-		NextPageToken:  nextPageToken,
+	return &pb.ListMutationsResponse{
+		Mutations:     mutations,
+		NextPageToken: nextPageToken,
 	}, nil
 }
 
-func (s *Server) logProofs(ctx context.Context, logID, firstTreeSize int64, epoch int64) (*tpb.GetLatestSignedLogRootResponse, *tpb.GetConsistencyProofResponse, *tpb.GetInclusionProofResponse, error) {
+// lowestSequenceNumber picks a lower bound on sequence number.
+// It returns the max between token and the high water mark of the previous epoch.
```

Comment:
are these three funcs pure moves and no edits along the way?

Reply:
ok